tex/gregoriotex.lua

```diff
@@ -120,6 +120,37 @@ local translation_mark = 1
 local abovelinestext_mark = 2
 log("marker whatsit id is %d", marker_whatsit_id)
 
+local function gregorio_exe()
```

Comment:
Since you've made this a function, doesn't this mean that this code block will be executed each time we need the program name?  Since that will always be the same for a given compilation, wouldn't it be more efficient to run the check code once when the package is loaded, defining a variable with the executable name in the process and then simply accessing that variable when we need it?